VMRecovery/Testing/Create-TestVM.ps1

```diff
@@ -0,0 +1,848 @@
+    $vmCreateTime = get-date ((get-date).ToUniversalTime()) -format yyyy-MM-ddTHH:mm:ss
+    if ($wait)
+    {
+        if ($disableBginfoExtension)
+        {
+            if ($customImage)
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -DisableBginfoExtension -Image $imageName
+            }
+            else
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -DisableBginfoExtension
+            }
+        }
+        else
+        {
+            if ($customImage)
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -Image $imageName
+            }
+            else
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm
+            }
+        }
+    }
+    else
+    {
+        if ($disableBginfoExtension)
+        {
+            if ($customImage)
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -DisableBginfoExtension -Image $imageName -AsJob
+            }
+            else
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -DisableBginfoExtension -AsJob
+            }
+        }
+        else
+        {
+            if ($customImage)
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -Image $imageName -AsJob
+            }
+            else
+            {
+                $result = New-AzureRmVM -ResourceGroupName $resourceGroupName -Location $location -VM $vm -AsJob
+            }
+        }
+    }
+
+    $vmdetail = [ordered]@{
+        'ResourceGroupName' = $resourceGroupName
+        'Name' = $vmName
+        'Location' = $location
+        'OsType' = $osType
+    }
+    $vmdetail = new-object -TypeName PSObject -Property $vmdetail
+    [void]$vmdetails.add($vmdetail)
+    $numCreated++
+    show-progress "$numCreated of $numToCreate VMs created"
+}
+
+if ($enableEMS)
+{
+    $vmdetails | where {$_.OSType -eq 'Windows'} | foreach {
+        show-progress "Enabling EMS on $($_.Name)"
+        $name = 'EnableEMS'
+        $publisher = 'Microsoft.Compute'
+        $extensionType = 'CustomScriptExtension'
+        $typeHandlerVersion = '1.9'
+        $settingString = '{"commandToExecute": "cmd.exe /c bcdedit /ems {current} on && bcdedit /emssettings EMSPORT:1 EMSBAUDRATE:115200 && shutdown /r /t 0 /f "}'
+        $result = Set-AzureRmVMExtension -AsJob -ResourceGroupName $_.resourceGroupName -VMName $_.Name -Location $_.Location -Name $name -Publisher $publisher -ExtensionType $extensionType -TypeHandlerVersion $typeHandlerVersion -settingstring $settingString
+    }
+}
+
+if ($trackLatency)
```

Comment:
Lets think about if there is a way to do MSFT internal actions such as these without publishing externally?  Seems like you are most of the way there as the get-kusto script is not published, there are some table names, etc being shown.  Also, what happens if someone external runs the script without this file?  Does it error out?

Reply:
The testing script for now we will hold off until we get this vetted and decide if we want to put it in this repo.